### DISCUSSION TITLE
Scalar product of two vectors in kernel

### DISCUSSION THREAD

**Discussion by Minjiang-Zhu** [idaholab/moose#23492] [Q&A General] [answered]
I wonder how to make scalar product of two vectors when writing a kernel, like u=v.*w, while all of u, v, w are vectors, or specifically, ADRealVectorValue. Is there a related documentation explaining this, or other related operations?

**Comment by GiudGiud** [comment, marked as answer]
You can have a look at the doxygen for the base class of RealVectorValue
https://mooseframework.inl.gov/docs/doxygen/libmesh/classlibMesh_1_1VectorValue.html

**Reply by GiudGiud** [reply]
and if you cant find that there, this is a very simple for loop

**Comment by hugary1995** [comment]
What you described is not a scalar product btw.
